### PR TITLE
Upgrade to 3.0

### PR DESCRIPTION
**Pull request by t0yv0**
The first commit is mechanical bump to 3.x and 4.x aws.

However we run into removal of `pulumi.ProviderMain` from upstream, what should we use instead?

```
../../pkg/provider/provider.go:23:38: undefined: pulumi.ProviderArgs
../../pkg/provider/provider.go:24:9: undefined: pulumi.ProviderArgs
../../pkg/provider/provider.go:43:68: undefined: pulumi.ConstructInputs
../../pkg/provider/provider.go:44:34: undefined: pulumi.ConstructResult
../../pkg/provider/provider.go:50:10: undefined: pulumi.ConstructResult
../../pkg/provider/provider.go:56:10: undefined: pulumi.ConstructResult
../../pkg/provider/provider.go:60:9: undefined: pulumi.ConstructResult
../../pkg/provider/serve.go:24:12: undefined: pulumi.ProviderMain
```

### CODE REVIEW

**Comment by t0yv0** [comment]
There is `pkg/resource/provider` I found as a candidate:

```
// Main is the typical entrypoint for a resource provider plugin.  Using it isn't required but can cut down
// significantly on the amount of boilerplate necessary to fire up a new resource provider.
func Main(name string, provMaker func(*HostClient) (pulumirpc.ResourceProviderServer, error)) error {
```

**Comment by justinvp** [comment]
Ah, yes, we need to update to the latest API. I can push a commit to this branch.

**Comment by t0yv0** [comment]
OK, yes please, that may save me a goose chase.

**Comment by justinvp** [comment]
I've pushed a commit to this branch.

File: provider/pkg/provider/serve.go [hunk 1]
```diff
@@ -15,13 +15,14 @@
...
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
-	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
 )
 
+// Serve launches the gRPC server for the resource provider.
 func Serve(providerName, version string, schema []byte) {
-	args := makeProviderArgs(schema)
-	if err := pulumi.ProviderMain(providerName, version, args); err != nil {
+	// Start gRPC service.
+	if err := provider.ComponentMain(providerName, version, schema, construct); err != nil {
```
Comment (t0yv0) [inline provider/pkg/provider/serve.go:25]: 👍

**Comment by justinvp** [comment]
I'm going to push a commit to this PR that adopts the latest change in master from 6811.

**Review by justinvp — APPROVED** [review]
LGTM

**Pull request merged**